src/controllers/authController.ts

```diff
@@ -0,0 +1,95 @@
+import jwt, { JwtPayload, TokenExpiredError } from 'jsonwebtoken';
+import bcrypt from 'bcryptjs';
+import { validationResult } from 'express-validator';
+import { Request, Response } from 'express';
+import { Types } from 'mongoose';
+
+import User from '../models/user.js';
+import Role from '../models/role.js';
+import { secret } from '../config/tokenKey.js';
+
+const generateAccessToken = (id: Types.ObjectId, roles: string[]) => {
+    const payload = {
+        id,
+        roles,
+    };
+    return jwt.sign(payload, secret, { expiresIn: '120000ms' });
+};
+
+class authController {
+    async registration(req: Request, res: Response) {
+        try {
```

Comment:
create errorHandlingMiddleware that wraps the controller method call in a try/catch statement. That allows to you remove all try/catch statements from controller methods.
example errorHandlingMiddleware:

 ```
 function (req: Request, res: Response, next: NextFunction) {
   try{
       next()
    }
    catch(e){
     res.status(400).json({ message: e.message });
     }
}
```